include/swift/AST/Attr.h

```diff
@@ -2145,19 +2145,29 @@ class TransposeAttr final
 /// has a different name).
 class HasAsyncAlternativeAttr final : public DeclAttribute  {
 public:
-  /// An optional name of the async alternative function, where the name of the
-  /// attributed function is used otherwise.
-  const DeclNameRef Name;
 
-  HasAsyncAlternativeAttr(DeclNameRef Name, SourceLoc AtLoc, SourceRange Range)
-    : DeclAttribute(DAK_HasAsyncAlternative, AtLoc, Range, false),
-      Name(Name) {}
+  /// Source location of the async function name in the attribute
+  const SourceLoc AsyncFunctionNameLoc;
 
-  HasAsyncAlternativeAttr(SourceLoc AtLoc, SourceRange Range)
-    : DeclAttribute(DAK_HasAsyncAlternative, AtLoc, Range, false) {}
+  /// DeclName of the async function in the attribute
+  const DeclNameRef AsyncFunctionName;
 
-  /// Determine whether this attribute has a name associated with it.
-  bool hasName() const { return !Name.getBaseName().empty(); }
+  /// Source location of the completion handler index passed to the index
+  const SourceLoc CompletionHandlerIndexLoc;
+
+  /// The index of the completion handler
+  const uint64_t CompletionHandlerIndex;
```

Comment:
ParameterList uses `size_t`, may as well use it here as well (even if they are the same)